apps/api/src/modules/notifications/queues/queue.service.ts

```diff
@@ -0,0 +1,179 @@
+import { ChannelType } from 'src/common/constants/notifications';
+import { MailgunNotificationConsumer } from 'src/jobs/consumers/notifications/mailgun-notifications.job.consumer';
+import { SmsKapsystemNotificationsConsumer } from 'src/jobs/consumers/notifications/smsKapsystem-notifications.job.consumer';
+import { SmsPlivoNotificationsConsumer } from 'src/jobs/consumers/notifications/smsPlivo-notifications.job.consumer';
+import { SmsTwilioNotificationsConsumer } from 'src/jobs/consumers/notifications/smsTwilio-notifications.job.consumer';
+import { SmtpNotificationConsumer } from 'src/jobs/consumers/notifications/smtp-notifications.job.consumer';
+import { Wa360dialogNotificationsConsumer } from 'src/jobs/consumers/notifications/wa360dialog-notifications.job.consumer';
+import { WaTwilioNotificationsConsumer } from 'src/jobs/consumers/notifications/waTwilio-notifications.job.consumer';
+import { WaTwilioBusinessNotificationsConsumer } from 'src/jobs/consumers/notifications/waTwilioBusiness-notifications.job.consumer';
+
+@Injectable()
+export class QueueService {
+  private readonly logger = new Logger(QueueService.name);
+  private queues: Map<string, Queue> = new Map();
+  private workers: Map<string, Worker> = new Map();
+  private queueEvents: Map<string, QueueEvents> = new Map();
+  private redisConfig: { host: string; port: number };
+  private idleTimeout = 10 * 60 * 1000; // 10 minutes
+  private cleanupInterval = 5 * 60 * 1000; // 5 minutes
+
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly smtpNotificationConsumer: SmtpNotificationConsumer,
+    private readonly mailgunNotificationConsumer: MailgunNotificationConsumer,
+    private readonly wa360dialogNotificationConsumer: Wa360dialogNotificationsConsumer,
+    private readonly waTwilioNotificationConsumer: WaTwilioNotificationsConsumer,
+    private readonly smsTwilioNotificationConsumer: SmsTwilioNotificationsConsumer,
+    private readonly smsPlivoNotificationConsumer: SmsPlivoNotificationsConsumer,
+    private readonly waTwilioBusinessNotificationConsumer: WaTwilioBusinessNotificationsConsumer,
+    private readonly smsKapsystemNotificationConsumer: SmsKapsystemNotificationsConsumer,
+  ) {
+    this.redisConfig = {
+      host: this.configService.get<string>('REDIS_HOST'),
+      port: this.configService.get<number>('REDIS_PORT'),
+    };
+    this.startCleanupTask();
+  }
+
+  private createQueue(queueName: string): Queue {
+    const queue = new Queue(queueName, { connection: this.redisConfig });
+
+    queue.on('error', (error) => {
+      this.logger.error(`Redis connection error in queue ${queueName}:`, error);
+    });
+
+    this.queues.set(queueName, queue);
+
+    // Create QueueEvents for handling stalled jobs
+    const queueEvents = new QueueEvents(queueName, { connection: this.redisConfig });
+    queueEvents.on('stalled', async (jobId) => {
+      this.logger.warn(`Job stalled in queue ${queueName}, jobId: ${jobId}`);
```

Comment:
We Probably won't get this state. But Can you add a TODO comment instead to handle this. Or in Retry PR, you can handle this.

Reply:
Updated the comment to a TODO in https://github.com/OsmosysSoftware/osmo-x/pull/264/commits/2ca9aa862c4766cf2feac5836625c1e27fcb3281

I will check if I can handle this in the retry PR